be/src/exec/pipeline/crossjoin/cross_join_left_operator.h

```diff
@@ -0,0 +1,152 @@
+              _probe_column_count(probe_column_count),
+              _build_column_count(build_column_count),
+              _conjunct_ctxs(conjunct_ctxs),
+              _cross_join_context(cross_join_context) {}
+
+    ~CrossJoinLeftOperator() override = default;
+
+    void _copy_joined_rows_with_index_base_build(vectorized::ChunkPtr& chunk, size_t row_count, size_t probe_index,
+                                                 size_t build_index);
+    void _copy_joined_rows_with_index_base_probe(vectorized::ChunkPtr& chunk, size_t row_count, size_t probe_index,
+                                                 size_t build_index);
+
+    void _copy_probe_rows_with_index_base_build(vectorized::ColumnPtr& dest_col, vectorized::ColumnPtr& src_col,
+                                                size_t start_row, size_t copy_number);
+    void _copy_probe_rows_with_index_base_probe(vectorized::ColumnPtr& dest_col, vectorized::ColumnPtr& src_col,
+                                                size_t start_row, size_t copy_number);
+
+    void _copy_build_rows_with_index_base_build(vectorized::ColumnPtr& dest_col, vectorized::ColumnPtr& src_col,
+                                                size_t start_row, size_t row_count);
+    void _copy_build_rows_with_index_base_probe(vectorized::ColumnPtr& dest_col, vectorized::ColumnPtr& src_col,
+                                                size_t start_row, size_t row_count);
+
+    void _init_chunk(vectorized::ChunkPtr* chunk);
+    Status prepare(RuntimeState* state) override;
+
+    Status close(RuntimeState* state) override;
+
+    bool has_output() const override { return _probe_chunk != nullptr; }
+
+    bool need_input() const override;
+
+    bool is_finished() const override { return _is_finished && _probe_chunk == nullptr; }
+
+    void finish(RuntimeState* state) override { _is_finished = true; }
+
+    bool is_ready() const override;
+
+    StatusOr<vectorized::ChunkPtr> pull_chunk(RuntimeState* state) override;
+
+    Status push_chunk(RuntimeState* state, const vectorized::ChunkPtr& chunk) override;
+
+private:
+    // previsou saved chunk.
+    vectorized::ChunkPtr _pre_output_chunk = nullptr;
+
+    // used when scan rows in [0, _build_rows_threshold) rows.
+    size_t _within_threshold_build_rows_index = 0;
+    // used when scan rows in [_build_rows_threshold, _number_of_built_rows) rows.
+    size_t _beyond_threshold_build_rows_index = 0;
+
+    // used as left table's chunk.
+    // _probe_chunk about one chunk_size(maybe 4096) of left table.
+    vectorized::ChunkPtr _probe_chunk = nullptr;
+    // used when scan chunks in right table based on one row of left table.
+    //  _probe_chunk_index is a local index in _probe_chunk.
+    size_t _probe_chunk_index = 0;
+    // used when scan rows in right table.
+    // _probe_rows_index is a local index in _probe_chunk.
+    // And is used when _probe_chunk_index == _probe_chunk->num_rows().
+    size_t _probe_rows_index = 0;
```

Comment:
Change the name of those two fields, here are my suggestion(only for reference, you can make a better one)
1. _probe_chunk_index -> _within_threshold_probe_rows_index
2. _probe_rows_index -> _beyond_threshold_probe_rows_index

Furthermore, could we combine these two fileds into one filed. Since the right table has been split into two parts, there is no need to split the cursor on the left. You can add TODO first